### PR TITLE
Add, test method to remove force data from a gp

### PR DESCRIPTION
**Pull request by stevetorr**
In the event of having a model that has bad training data you'd no longer like to use. Removes by internal index of force data; re-computes matrices, updates the internal training label np arrays, and also updates the global data labels. This addition is 100% covered by a unit test which starts with dummy data, makes predictions, adds data, makes new predictions, removes the data, and ensures that the changes this made to the GP were fully reversed.

### CODE REVIEW

File: flare/gp.py [hunk 1]
```diff
@@ -823,7 +834,48 @@ def adjust_cutoffs(self, new_cutoffs: Union[list, tuple, 'np.ndarray'],
...
+        # Listify input even if one integer
+        if isinstance(indexes, int):
+            indexes = [indexes]
+
+        if max(indexes) > len(self.training_data):
+            raise ValueError("Index out of range of data")
+
+        # Get in reverse order so that modifying higher indexes doesn't affect
+        # lower indexes
+        indexes.sort(reverse=True)
```
Comment (nw13slx) [inline flare/gp.py:866]: That's clever.

File: flare/gp.py [hunk 2]
```diff
@@ -823,7 +834,48 @@ def adjust_cutoffs(self, new_cutoffs: Union[list, tuple, 'np.ndarray'],
...
 
+        for i in indexes:
+            del self.training_data[i]
+            del self.training_labels[i]
+
+        self.training_labels_np = np.hstack(self.training_labels)
+        _global_training_data[self.name] = self.training_data
+        _global_training_labels[self.name] = self.training_labels_np
```
Comment (nw13slx) [inline flare/gp.py:874]: The self.all_labels may also need to be updated.

File: flare/gp.py [hunk 3]
```diff
@@ -823,7 +834,48 @@ def adjust_cutoffs(self, new_cutoffs: Union[list, tuple, 'np.ndarray'],
...
+
+        if update_matrices:
+            self.set_L_alpha()
+            self.compute_matrices()
```
Comment (nw13slx) [inline flare/gp.py:879]: maybe consider returning the deleted data and labels like a pop function.
